Add documentation on crawl-configuration

laanwj opened the pull request:
- Tried to document the new crawl configuration in `crawl-configuration.md`.
- Added a new directory for detailed documentation (`doc`) where documents about advanced functionality such as  e.g. mass scanning can be placed, and added references to this.
- Moved the suggestion sections in README.md under a parent section "What is scanned for?".

Please let me know if I'm on the right track with this.

s-rah commented:
> Please let me know if I'm on the right track with this.

Thanks, this is awesome! 

Thoughts on the mid-term documentation effort: The front README.md needs a complete overhaul - the nature of OnionScan has shifted from web based stuff to all kinds of services and correlations - so the section about the kinds of scans it does isn't up to date anymore. Eventually I would image that some of the stuff from: https://github.com/s-rah/onionscan/blob/onionscan-0.2/design/000-onionscan.md combined with some of the pictures from the various OnionScan reports will make it into the README - we can definitely be a lot more informative about what OnionScan can be used for :)

I'm happy with `/doc` for placing more advanced stuff. Did you have more you wanted to do to this PR or should I merge it in? I think this is definitely a step in the right direction.

laanwj commented:
> The front README.md needs a complete overhaul - the nature of OnionScan has shifted from web based stuff to all kinds of services and correlations - so the section about the kinds of scans it does isn't up to date anymore

Right, it covers very little ground right now, and inside the main README.md there's not much scope for it to grow. I've added a commit that factors out the list of detected problems into a separate document, and makes clear that these are web issues.

> I would image that some of the stuff from: https://github.com/s-rah/onionscan/blob/onionscan-0.2/design/000-onionscan.md combined with some of the pictures from the various OnionScan reports will make it into the README

Absolutely, I'd imagine the blurb from there would be helpful to describe the software in a general introduction. It could be used almost as-is in the README, just aimed at users instead of developers.

I've added a commit that adds a "What is OnionScan?" section with a bit of this information. I wasn't entirely clear how to express the twofold purpose so I kept it aimed at onion service providers (which are probably still the main audience).

> I'm happy with /doc for placing more advanced stuff. Did you have more you wanted to do to this PR or should I merge it in? I think this is definitely a step in the right direction.

Ok, great! Yes if this is a step in the right direction let's merge it so that the structure is in place, and then improve/add more later.
